solr/core/src/java/org/apache/solr/core/PluginBag.java

```diff
@@ -288,6 +294,20 @@ public PluginHolder<T> put(String name, PluginHolder<T> plugin) {
     return old;
   }
 
+  static final class APIConfigProviderBinder extends AbstractBinder {
```

Comment:
[-0] It might be better to have this live as a public class in `org.apache.solr.jersey`.  I couldn't avoid dragging some JAX-RS/api logic into PluginBag, but IMO the less that lives in here the better.